### PR TITLE
Fix getCountForPagination for grouped query

### PR DESCRIPTION
**Pull request by codelight89**
As discussed here: https://github.com/laravel/framework/pull/8113

### CODE REVIEW

**Comment by codelight89** [comment]
Its silence in original pull request. So I decided to make it.

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1566,65 +1576,58 @@ public function exists()
...
-		return is_array($count) ? count($count) : (int) $count;
+		return (int) $this->aggregate(__FUNCTION__, $columns);
 	}
 
 	/**
 	 * Retrieve the minimum value of a given column.
 	 *
 	 * @param  string  $column
-	 * @return float|int
+	 * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: no!

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1566,65 +1576,58 @@ public function exists()
...
-		return is_array($result) ? min($result) : $result;
+		return $this->aggregate(__FUNCTION__, array($column));
 	}
 
 	/**
 	 * Retrieve the maximum value of a given column.
 	 *
 	 * @param  string  $column
-	 * @return float|int
+	 * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: no!

File: src/Illuminate/Database/Query/Builder.php [hunk 3]
```diff
@@ -1566,65 +1576,58 @@ public function exists()
...
-		return is_array($result) ? max($result) : $result;
+		return $this->aggregate(__FUNCTION__, array($column));
 	}
 
 	/**
 	 * Retrieve the sum of the values of a given column.
 	 *
 	 * @param  string  $column
-	 * @return float|int
+	 * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: no!

File: src/Illuminate/Database/Query/Builder.php [hunk 4]
```diff
@@ -1566,65 +1576,58 @@ public function exists()
...
-		return is_array($result) ? array_sum($result) : $result ?: 0;
+		return $result ?: 0;
 	}
 
 	/**
 	 * Retrieve the average of the values of a given column.
 	 *
 	 * @param  string  $column
-	 * @return float|int
+	 * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: no!

**Comment by codelight89** [comment]
@GrahamCampbell, sorry, I forget to fix this after reverting old code :smile:

**Comment by GrahamCampbell** [comment]
I can't merge this without tests.

**Comment by codelight89** [comment]
@GrahamCampbell But I wrote the test

**Comment by GrahamCampbell** [comment]
> But I wrote the test

This pull request deletes the tests?

File: src/Illuminate/Database/Query/Builder.php [hunk 5]
```diff
@@ -1420,17 +1420,27 @@ public function simplePaginate($perPage = 15, $columns = ['*'])
 	/**
 	 * Get the count of the total records for the paginator.
 	 *
+	 * @param  array  $columns
 	 * @return int
 	 */
-	public function getCountForPagination()
+	public function getCountForPagination($columns = ['*'])
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: neeeds a typehint

**Comment by codelight89** [comment]
> This pull request deletes the tests?

only tests added in previous commits of this pull request

**Pull request merged**